Source/Core/Resource.js

```diff
@@ -126,6 +130,34 @@ define([
         request.deferred = undefined;
     }
 
+    /**
+     * @private
+     */
+    function combineQueryParameters(q1, q2, preserveQueryParameters) {
+        if (!preserveQueryParameters) {
+            return combine(q1, q2);
+        }
+
+        var result = clone(q1, true);
+        for (var param in q2) {
+            if (q2.hasOwnProperty(param)) {
+                var value = result[param];
+                var q2Value = q2[param];
+                if (defined(value)) {
+                    if (!Array.isArray(value)) {
```

Comment:
Would it make things simpler to have the query properties always be arrays?

Reply:
Possibly, however we would need to change functionality of `objectToQuery` and `queryToObject` to do this. Then a user could still `queryParameters` property and change it manually, so we'd have to implement a mechanism to prevent that. Since its already done I think it'll be more trouble than its worth.